pkg/utils/tekton/controller.go

```diff
@@ -206,3 +206,88 @@ func findTagWithName(tags *unstructured.UnstructuredList, name string) *unstruct
 
 	return nil
 }
+
+func (k KubeController) CreateOrUpdateSigningSecret(publicKey []byte, name, namespace string) (err error) {
+	api := k.Tektonctrl.K8sClient.KubeInterface().CoreV1().Secrets(namespace)
+	ctx := context.TODO()
+
+	expectedSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: name},
+		Data:       map[string][]byte{"cosign.pub": publicKey},
+	}
+
+	s, err := api.Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return
+		}
+		if _, err = api.Create(ctx, expectedSecret, metav1.CreateOptions{}); err != nil {
+			return
+		}
+	} else {
+		if string(s.Data["cosign.pub"]) != string(publicKey) {
+			if _, err = api.Update(ctx, expectedSecret, metav1.UpdateOptions{}); err != nil {
+				return
+			}
+		}
+	}
+	return
+}
+
+func (k KubeController) GetPublicKey(name, namespace string) (publicKey []byte, err error) {
+	api := k.Tektonctrl.K8sClient.KubeInterface().CoreV1().Secrets(namespace)
+	ctx := context.TODO()
+
+	secret, err := api.Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+
+	publicKey = secret.Data["cosign.pub"]
+	return
+}
+
+const configPolicyName = "ec-policy"
```

Comment:
if this is a constant put this on the top of the file or make use of the constants pkg